### DISCUSSION TITLE
[twig-component] How to get block values in PHP

### DISCUSSION THREAD

**Discussion by tacman** [symfony/ux#2139] [Questions & Answers]
I have an accordion component that can receive data as either attributes or blocks.  I'm wondering if there's a more elegant way of gettng the block content.

```twig
                        {# as attributes #}
                        <twig:accordion :open="open"
                                        :header="company_jobTitle()"
                                        :body="lorem_sentences(40, true)"
                        >
                        </twig:accordion>

                        {# as blocks #}
                        <twig:accordion :open="open"
                        >
                            <twig:block name="header">
                                {{ company_jobTitle() }}
                            </twig:block>
                            <twig:block name="body">
                                {{ lorem_sentences(40, true) }}
                            </twig:block>
                        </twig:accordion>
````

To render, I have to repeat come code:

```html
<h2 class="accordion-header" id="heading{{ id }}">
    <button type="button" class="accordion-button"
            data-bs-toggle="collapse"
            data-bs-target="#accordion{{ id }}" aria-expanded="true" aria-controls="accordion{{ id }}">
        {{ header|default(block('header') is defined ? block('header') : '!!') }}
    </button>
</h2>

<div id="accordion{{ id }}" class="accordion-collapse collapse {{ open ? 'show' }}" {{ bsParent ? 'data-bs-parent="%s"'|format(bsParent) }}>
    <div class="accordionbody">
        {% block body %}
        {{ body|default(block('body') is defined ? block('body') : '!!')  }}
        {% endblock %}
    </div>
</div>
```

I'd like to get the value of header and body inside the PHP part of the component.  In PreMount, I have access to the data _attributes_ passed in.  Is there any way to access the block values in the PHP?  Or a more elegant way to do this?

In particular, since the ID is created from the header, if the header comes in as a block I have to create the id in twig, unless I can get the block value some other way.  

It'd be cool if the block command could take default if it didn't exist.

```twig
{{ block('someBlockName', default='default-value if block doesn't exist') }}
```

**Comment by leevigraham** [comment]
I've been trying to figure out the same thing (but for an Alert component).

I don't think you can access the twig template in the PHP class because it hasn't been mounted yet.

> `{{ block('someBlockName', 'default-value if block doesn't exist') }}`

FYI The block function already has a second argument `template`: https://twig.symfony.com/doc/3.x/functions/block.html

**Reply by tacman** [reply]
> FYI The block function already has a second argument

Good point, I've updated my post to reflect a new parameter, default.
